legend-engine-xts-data-space/legend-engine-xt-data-space-grammar/src/main/java/org/finos/legend/engine/language/pure/grammar/from/DataSpaceParseTreeWalker.java

```diff
@@ -220,6 +231,54 @@ private DataSpaceExecutable visitDataSpaceExecutable(DataSpaceParserGrammar.Exec
+
+        // Name
+        DataSpaceParserGrammar.ExecutableTitleContext executableTitleContext = PureGrammarParserUtility.validateAndExtractRequiredField(ctx.executableTitle(), "title", executable.sourceInformation);
+        executable.title = PureGrammarParserUtility.fromGrammarString(executableTitleContext.STRING().getText(), true);
+
+        // Description (optional)
+        DataSpaceParserGrammar.ExecutableDescriptionContext descriptionContext = PureGrammarParserUtility.validateAndExtractOptionalField(ctx.executableDescription(), "description", executable.sourceInformation);
+        executable.description = descriptionContext != null ? PureGrammarParserUtility.fromGrammarString(descriptionContext.STRING().getText(), true) : null;
+
+        // query
+        DataSpaceParserGrammar.ExecutableTemplateQueryContext queryContext = PureGrammarParserUtility.validateAndExtractRequiredField(ctx.executableTemplateQuery(), "query", executable.sourceInformation);
+        executable.query = visitLambda(queryContext.combinedExpression());
+
+        // executionContextKey
+        DataSpaceParserGrammar.ExecutableExecutionContextKeyContext executionContextKeyContext = PureGrammarParserUtility.validateAndExtractOptionalField(ctx.executableExecutionContextKey(), "executionContextKey", executable.sourceInformation);
+        executable.executionContextKey = executionContextKeyContext != null ? PureGrammarParserUtility.fromGrammarString(executionContextKeyContext.STRING().getText(), true) : null;
+
+        return executable;
+    }
+
+    private Lambda visitLambda(DataSpaceParserGrammar.CombinedExpressionContext ctx)
+    {
+        DomainParser parser = new DomainParser();
+        // prepare island grammar walker source information
+        int startLine = ctx.getStart().getLine();
+        int lineOffset = walkerSourceInformation.getLineOffset() + startLine - 1;
+        // only add current walker source information column offset if this is the first line
+        int columnOffset = (startLine == 1 ? walkerSourceInformation.getColumnOffset() : 0) + ctx.getStart().getCharPositionInLine();
+        ParseTreeWalkerSourceInformation combineExpressionSourceInformation = new ParseTreeWalkerSourceInformation.Builder(walkerSourceInformation.getSourceId(), lineOffset, columnOffset).withReturnSourceInfo(this.walkerSourceInformation.getReturnSourceInfo()).build();
+        String lambdaString = this.input.getText(new Interval(ctx.start.getStartIndex(), ctx.stop.getStopIndex()));
+        ValueSpecification valueSpecification = parser.parseCombinedExpression(lambdaString, combineExpressionSourceInformation, null);
+        if (valueSpecification instanceof Lambda)
+        {
+            return (Lambda) valueSpecification;
+        }
+        // NOTE: If the user just provides the body of the lambda, we will wrap a lambda around it
+        // we might want to reconsider this behavior and throw error if this convenience causes any trouble
```

Comment:
Can we add a test for this? I couldn't see a test in this commit that exercises this flow of wrapping the body in a lambda. 
Will this wrapping break grammar roundtrip?

Reply:
Hello Aziem, thanks for your review. 🥹🥹
 It's the same logic as how Service handles Lambda.  🥹🫣😅👀
The roundtrip will append a `|`  for users. If you compare this 'problematic' string with the string returned from the roundtrip, they will be different.